test/test_unary_ufuncs.py

```diff
@@ -450,6 +450,38 @@ def test_out_arg_all_dtypes(self, device, dtype, op):
             else:
                 self._test_out_arg(op, input, out)
 
+    @ops(filter(lambda op: op.supports_dtype_kwarg, unary_ufuncs), dtypes=OpDTypes.supported)
+    def test_dtype_arg(self, device, dtype, op):
+        input = make_tensor((32, 32), dtype=dtype, device=device,
+                            low=op.domain[0], high=op.domain[1])
+
+        for computation_dtype in all_types_and_complex_and(torch.bool, torch.half):
+            if computation_dtype in [torch.float16]:
+                continue
+            # As NumPy doesn't have a bfloat16 equivalent.
+            if torch.bfloat16 in [computation_dtype, input.dtype]:
+                continue
+
+            np_computation_dtype = torch_to_numpy_dtype_dict[computation_dtype]
+            try:
+                expected = op.ref(input.numpy(), dtype=np_computation_dtype)
+            except TypeError:
+                expected = None
+
+            if expected is None:
+                # Verify that torch variant raises an error.
+                if dtype.is_complex and computation_dtype.is_floating_point:
+                    # Pytorch doesn't raise an error for
```

Comment:
Maybe we should raise this error when using dtype

Reply:
Currently it does raise a warning.